test/unit/codegen/codegen_llvm_ir.cpp

```diff
@@ -1436,3 +1443,51 @@ SCENARIO("Dead code removal", "[visitor][llvm][opt]") {
         }
     }
 }
+
+//=============================================================================
+// Inlining: remove inline code blocks
+//=============================================================================
+
+SCENARIO("Removal of inlined functions and procedures", "[visitor][llvm][inline]") {
+    GIVEN("Simple breakpoint block calling a function and a procedure") {
+        std::string nmodl_text = R"(
+            NEURON {
+                SUFFIX test_inline
+                RANGE a, b, s
+            }
+            ASSIGNED {
+                a
+                b
+                s
+            }
+            PROCEDURE test_add(a, b) {
+                LOCAL i
+                i = a + b
+            }
+            FUNCTION test_sub(a, b) {
+                test_sub = a - b
+            }
+            BREAKPOINT {
+                SOLVE states METHOD cnexp
+            }
+            DERIVATIVE states {
+                a = 1
+                b = 2
+                test_add(a, b)
+                s = test_sub(a, b)
+            }
+        )";
+
+        THEN("when the code is inlined the procedure and function blocks are removed") {
+            std::string module_string =
+                run_llvm_visitor(nmodl_text, false, false, 1, "none", {}, true);
```

Comment:
nit: maybe can add `/*field=*/`-like comments?

Reply:
Good point, I forgot it :facepalm: